tests/python/relay/test_pass_qnn_legalize.py

```diff
@@ -296,7 +297,98 @@ def _get_mod(data_dtype, kernel_dtype):
         assert "cast" in legalized_mod.astext() and "qnn" in legalized_mod.astext()
 
 
+def test_qnn_legalize_qnn_conv2d_non_scalar_qnn_params():
+    """
+    Test QNN legalization for qnn.dense op for Hexagon target when kernel zero point and kernel
+    scale are not scalars.
```

Comment:
So, by "not scalar" I mean constant vector of scalars. For example:
`relay.const([val1, val2 ... valN])`
This is more interesting case for testing because on legalization we do padding of weights. And dimension of weights, vector of scale/zero point and 'channels' attribute should be aligned.